loopin.go

```diff
@@ -923,10 +960,23 @@ func (s *loopInSwap) waitForSwapComplete(ctx context.Context,
 				invoiceFinalized = true
 				htlcKeyRevealed = s.tryPushHtlcKey(ctx)
 
+				// Notify the main goroutine that the swap
+				// is finalized from the client's perspective. A
+				// new swap can be started while this goroutine
+				// is waiting for processing the on-chain spend.
+				invFinalizedChan <- invoiceFinalized
```

Comment:
There is a rare case this could deadlock. If this goroutine is already in this `case` but the parent context is canceled then we may not ever try to read this channel. I recommend putting this in a select like this:

```
select {
       case invFinalizedChan <- invoiceFinalized:
       case <-ctx.Done():
}
```

Reply:
Added this. I thought now where we manually cancel the parent context once the swap is completed we don't need this. But the parent context could also be canceled in the `subscribeAndLogUpdates` method.